packages/valory/skills/abstract_round_abci/tests/data/dummy_abci/behaviours.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains round behaviours of DummyAbciApp."""
+
+from typing import Generator, Set, Type, cast
+
+from packages.valory.skills.abstract_round_abci.base import AbstractRound
+from packages.valory.skills.abstract_round_abci.behaviours import (
+    AbstractRoundBehaviour,
+    BaseBehaviour,
+)
+from packages.valory.skills.abstract_round_abci.common import RandomnessBehaviour
+from packages.valory.skills.abstract_round_abci.tests.data.dummy_abci.models import (
+    Params,
+)
+from packages.valory.skills.abstract_round_abci.tests.data.dummy_abci.payloads import (
+    DummyFinalPayload,
+    DummyKeeperSelectionPayload,
+    DummyRandomnessPayload,
+    DummyStartingPayload,
+)
+from packages.valory.skills.abstract_round_abci.tests.data.dummy_abci.rounds import (
+    DummyAbciApp,
+    DummyFinalRound,
+    DummyKeeperSelectionRound,
+    DummyRandomnessRound,
+    DummyStartingRound,
+    SynchronizedData,
+)
+
+
+class DummyBaseBehaviour(BaseBehaviour):
+    """Base behaviour for the common apps' skill."""
+
+    @property
+    def synchronized_data(self) -> SynchronizedData:
+        """Return the synchronized data."""
+        return cast(SynchronizedData, super().synchronized_data)
+
+    @property
+    def params(self) -> Params:
+        """Return the params."""
+        return cast(Params, super().params)
+
+
+class DummyStartingBehaviour(DummyBaseBehaviour):
+    """DummyStartingBehaviour"""
+
+    behaviour_id: str = "dummy_starting"
+    matching_round: Type[AbstractRound] = DummyStartingRound
+
+    def async_act(self) -> Generator:
+        """Do the act, supporting asynchronous execution."""
+
+        with self.context.benchmark_tool.measure(self.behaviour_id).local():
+            content = True
+            sender = self.context.agent_address
+            payload = DummyStartingPayload(sender=sender, content=content)
+
+        with self.context.benchmark_tool.measure(self.behaviour_id).consensus():
+            yield from self.send_a2a_transaction(payload)
+            yield from self.wait_until_round_end()
+
+        self.set_done()
```

Comment:
You could move this to the `DummyBaseBehaviour` since it is repeated below too.

Reply:
I love it! However it may be modified for usage in other tests, lets do that at the end if not needed